### PR TITLE
Replace padded prop on withScrim with more generic contentClassName

### PR DESCRIPTION
**Pull request by danrr**
This allows the extraction of some modal specific code and makes the component more versatile.
Only the modal needs display: flex and padding on the container, it now injects the styles it requires.

### CODE REVIEW

File: packages/bpk-scrim-utils/src/bpk-scrim-content.scss [hunk 1]
```diff
@@ -25,26 +25,10 @@
   bottom: 0;
   left: 0;
   z-index: $bpk-zindex-scrim;
-  overflow: auto;
```
Comment (matthewdavidson) [inline packages/bpk-scrim-utils/src/bpk-scrim-content.scss]: Why is this removed?

Comment (danrr) [reply packages/bpk-scrim-utils/src/bpk-scrim-content.scss]: overflow-x and overflow-y on the next two lines overwrote it.

**Comment by danrr** [comment]
I had look on browser stack for iPhone and Android and locally in a bunch of browsers.

File: packages/bpk-scrim-utils/src/bpk-scrim-content.scss (hunk 1 above)
Comment (matthewdavidson) [reply packages/bpk-scrim-utils/src/bpk-scrim-content.scss]: Can we keep it in for now? I remember explicitly adding this for x-browser compat - I should have added a comment explaining why :(
